crates/subspace-farmer/src/cluster/nats_client.rs

```diff
@@ -0,0 +1,433 @@
+        } else {
+            Subject::from_static(Request::SUBJECT)
+        };
+        let message = self
+            .client
+            .request(subject.clone(), request.encode().into())
+            .await?;
+
+        let response =
+            Request::Response::decode(&mut message.payload.as_ref()).map_err(|error| {
+                warn!(
+                    %subject,
+                    %error,
+                    response_type = %type_name::<Request::Response>(),
+                    response = %hex::encode(message.payload),
+                    "Response decoding failed"
+                );
+
+                RequestErrorKind::Other
+            })?;
+
+        Ok(response)
+    }
+
+    /// Make request that expects stream response
+    pub async fn stream_request<Request>(
+        &self,
+        request: Request,
+        instance: Option<&str>,
+    ) -> Result<StreamResponseSubscriber<Request::Response>, StreamRequestError>
+    where
+        Request: GenericStreamRequest,
+    {
+        let stream_request = StreamRequest::new(request);
+
+        let subscriber = self
+            .client
+            .subscribe(stream_request.response_subject.clone())
+            .await?;
+
+        let subject = if let Some(instance) = instance {
+            Subject::from(Request::SUBJECT.replace('*', instance))
+        } else {
+            Subject::from_static(Request::SUBJECT)
+        };
+
+        self.client
+            .publish(subject, stream_request.encode().into())
+            .await?;
+
+        Ok(StreamResponseSubscriber {
+            subscriber,
+            buffered_responses: GenericStreamResponses::Continue(VecDeque::new()),
+            _phantom: PhantomData,
+        })
+    }
+
+    /// Make notification without waiting for response
+    pub async fn notification<Notification>(
+        &self,
+        notification: &Notification,
+        instance: Option<&str>,
+    ) -> Result<(), PublishError>
+    where
+        Notification: GenericNotification,
+    {
+        let subject = if let Some(instance) = instance {
+            Subject::from(Notification::SUBJECT.replace('*', instance))
+        } else {
+            Subject::from_static(Notification::SUBJECT)
+        };
+
+        self.client
+            .publish(subject, notification.encode().into())
+            .await
+    }
+
+    /// Send a broadcast message
+    pub async fn broadcast<Broadcast>(
+        &self,
+        message: &Broadcast,
+        instance: &str,
+    ) -> Result<(), PublishError>
+    where
+        Broadcast: GenericBroadcast,
+    {
+        self.client
+            .publish_with_headers(
+                Broadcast::SUBJECT.replace('*', instance),
+                {
+                    let mut headers = HeaderMap::new();
+                    if let Some(message_id) = message.deterministic_message_id() {
+                        headers.insert("Nats-Msg-Id", message_id);
```

Comment:
Is this the only usage of the string?

Reply:
Yes and it is a part of the headers that NATS understands: https://docs.nats.io/nats-concepts/jetstream/headers#publish